compiler/rustc_ast/src/token.rs

```diff
@@ -703,12 +704,18 @@ pub enum NonterminalKind {
 }
 
 impl NonterminalKind {
-    pub fn from_symbol(symbol: Symbol) -> Option<NonterminalKind> {
+    pub fn from_symbol(symbol: Symbol, span: Span) -> Option<NonterminalKind> {
```

Comment:
Could you pass an edition closure like in #80272 instead of the whole span?
(+ a comment telling that `span.edition()` is relatively expensive.)